File: sample-app-java/app/controllers/Wvyp.java
package controllers;

import data.ServiceClient;
import play.libs.F;
import play.mvc.Controller;
import play.mvc.Result;

import javax.inject.Inject;

public class Wvyp extends Controller {

  private final ServiceClient serviceClient;

  @Inject
  public Wvyp(ServiceClient serviceClient) {
    this.serviceClient = serviceClient;
  }

  public F.Promise<Result> index(Boolean embed) {
    F.Promise<String> wvypPromise = serviceClient.makeServiceCall("wvyp");
    F.Promise<String> searchPromise = serviceClient.makeServiceCall("search");

    return wvypPromise.flatMap(wvyp -> searchPromise.map(search -> {
      int wvypCount = Integer.parseInt(wvyp);
      int searchCount = Integer.parseInt(search);

      if (embed) {
        return ok(views.html.wvyp.wvypBody.apply(wvypCount, searchCount));
      } else {
        return ok(views.html.wvyp.wvyp.apply(wvypCount, searchCount));
      }
    }));
  }
}

File: sample-app/app/controllers/Wvyp.scala
package controllers

import play.api.mvc.{Cookie, Action, Controller}
import data.ServiceClient
import play.api.libs.concurrent.Execution.Implicits._

/**
 * An example standalone module that represents the "Who's Viewed Your Profile" module from the LinkedIn homepage.
 */
class Wvyp(serviceClient: ServiceClient) extends Controller {

  def index(embed: Boolean) = Action.async { implicit request =>
    val wvypCountFuture = serviceClient.makeServiceCall("wvyp")
    val searchCountFuture = serviceClient.makeServiceCall("search")

    for {
      wvypCount <- wvypCountFuture
      searchCount <- searchCountFuture
    } yield {
      if (embed) Ok(views.html.wvyp.wvypBody(wvypCount.toInt, searchCount.toInt)).withCookies(Cookie("foo", "bar"))
      else Ok(views.html.wvyp.wvyp(wvypCount.toInt, searchCount.toInt))
    }
  }
}
